homeassistant/components/intellifire/__init__.py

```diff
@@ -17,17 +20,35 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up IntelliFire from a config entry."""
     LOGGER.debug("Setting up config entry: %s", entry.unique_id)
 
-    # Define the API Object
-    api_object = IntellifireAsync(entry.data[CONF_HOST])
+    if "username" not in entry.data:
+        LOGGER.debug("Auth Failed: %s", entry.unique_id)
+        raise ConfigEntryAuthFailed
+
+    # Define the API Objects
+    read_object = IntellifireAsync(entry.data[CONF_HOST])
+    ift_control = IntellifireControlAsync(
+        fireplace_ip=entry.data[CONF_HOST],
+    )
+    try:
+        await ift_control.login(
+            username=entry.data[CONF_USERNAME],
+            password=entry.data[CONF_PASSWORD],
+        )
+    except (ConnectionError, ClientConnectionError) as err:
+        raise ConfigEntryNotReady from err
+    except (LoginException, KeyError) as err:
```

Comment:
KeyError feels a bit odd, is that to catch `CONF_PASSWORD` missing ?

Reply:
Yea it is -> because if its an old version of the config `CONF_PASSWORD` isn't there and then you get a key error..

Reply:
To elaborate -> The existing version doesn't have password/username ... and since we didn't need to do the migration route but just fire a ReAuth error if they are missing thats what this logic was for...

Reply:
but you guard for that already with checking for `CONF_USERNAME` a few lines above?

Reply:
You are 100% right ... I'll test to make sure Its not some random edge case and push a change

Reply:
Totally unnecessary -> good catch! :)

pushing/resolving